### PR TITLE
Chillout

### PR DESCRIPTION
**Pull request by mike1729**
Remove a validator from pallet staking when they get a ban.

- New feature (non-breaking change which adds functionality)

# Checklist:

- I have made corresponding changes to the existing documentation
- I have bumped `spec_version` and `transaction_version`

### CODE REVIEW

**Review by Marcin-Radecki — APPROVED** [review]

**Comment by Marcin-Radecki** [comment]
Actually one more comment: since it is possible that `reserved` validator ends up in `BanValidator` event, so we should not remove validators that are in the reserved nodes list.

**Review by Marcin-Radecki — APPROVED** [review]
Great contribution.

File: pallets/elections/src/traits.rs [hunk 1]
```diff
@@ -86,3 +86,21 @@ where
         pallet_staking::ErasStakers::<T>::iter_key_prefix(era).collect()
     }
 }
+
+pub trait ValidatorExtractor {
+    type AccountId;
+
+    /// Removes given validator from pallet's staking valiators list
```
Comment (DamianStraszak) [inline pallets/elections/src/traits.rs:93]: typo, should be -> Removes given validator from pallet's staking validator list

**Review by DamianStraszak — APPROVED** [review]

File: pallets/elections/src/traits.rs (hunk 1 above)
Comment (mike1729) [reply pallets/elections/src/traits.rs]: Fixed.

**Pull request merged**